File: 003.clj
; What is the largest prime factor of the number 600851475143 ?

(defn factor? [n d]
  (zero? (mod n d)))

(defn factors [n]
  (filter #(factor? n %) (range 2 n)))

(defn prime? [n]
  (empty? (factors n)))

(println (apply max (take-while prime? (factors 600851475143))))

File: 003.hs
-- What is the largest prime factor of the number 600851475143 ?

module Main where

factor n d
  | mod n d == 0 = True
  | otherwise = False

factors n = [d | d <- [2..n-1], mod n d == 0]

prime n = null (factors n)

main = putStrLn (show answer)
answer = maximum $ takeWhile prime (factors 600851475143)

-- 6857
